docs/getting-started/proofs-and-verification/proofs.md

```diff
@@ -44,4 +47,54 @@ message GetDataProofResponse {
 message MPTrieProofElement {
   repeated bytes hashes = 1;
 }
-```
\ No newline at end of file
+```
+
+### Proofs
+
+#### Ledger connectivity proof
+Proof generation:
+- Zero step – user provided with receipt (block header) from another user who claim that some block is part of ledger
+- First step – user requests header of last block in ledger, without discovering what block he wants to validate
+  - User can store genesis block apriori or request it as well
+- Second step – user requests the shortest path in skip list from block he validates to genesis block
+- The third step – user requests the shortest path from last block (first step) to block he validates
+- The forth step – user validates all apriori known headers - genesis, block, last - in both paths and concatenate both paths, result is block proof
+
+Proof validation:
+- For each block header in proof, starting from Genesis
+  - Calculate block hash, by calculating block header hash
+  - If calculated hash is in next block header hashes list, continue
+  - If not, fail
+
+Optimization - because result of each call to `Commit()` is transaction receipt contains block header, user can use this header, only make sure that it has higher block number
+
+Please note that algorithm above allows to build proofs from multiple blocks.
+For example, proof for blocks 3 and 5 will include (8, 6, 5, 4, 3, 2, 0). It is useful while validating specific value history.
+
+Algorithm above, used as first step of [transaction proof](#transaction-existence-proof)/[state proof](#state-proof).
+- To prove that block(s) is/are part of ledger
+
+
+#### Transaction existence proof
+Detailed explanation for transaction existence proof, based on data stored during tx commit
+Alice submit tx and in the future should prove it existence:
+- During transaction commit, `TxReceipt`, containing block header returned as proof.
+    - Transaction content and `TxReceipt` stored by Alice to use as proof to another users in the future.
+- At some point in future Alice need to prove (to Bob) that tx is part of ledger, using proof stored since tx commit
+    - Alice provides Bob with `TxReceipt` and transaction content.
+    - First, Bob checks ledger connectivity and that block header in `TxReceipt` is part of ledger, as described [here](#ledger-connectivity-proof)
+    - Second, Bob calls `GetTxProof()` for tx path in the block merkle tree and tx content already provided by Alice
+    - Bob validates proof using tx hash and merkle root stored in block header, for more details see [here](../../architecture-and-design/tx-merkle-tree#merkle-tree-proof-example)
```

Comment:
In the presentations we called these two pieces together the transaction evidence: TxReceipt signed by server + Transaction envelope signed by client (Alice)